text/0000-cache-types.md

```diff
@@ -0,0 +1,109 @@
+
+# What it is
+[what-it-is]: #what-it-is
+
+Buildpacks will provide a mechanism to denote a cached layer as something that can be shared by associating specifing a scope of "shared".
+
+- Target persona: buildpack author
+
+There will be two defined scopes: 
+
+- `app` - This cached layer is specific to the application and SHOULD NOT be shared with any other applications.
+- `shared` - This cached layer is specific to the buildpack and MAY be shared across multiple applications.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+#### Platform Implementation
+
+Platforms can decide how to _share_ the "shared" cached layers based on their specific use cases.
+
+In `pack`, it being a local development tool, shared cached layers could be implemented as a single global cache that is shared across all applications. Additional options can be provided to restrict or otherwise configure this sharing.
+
+In a different platform, one can foresee that the cache is shared only within a "project" or an "organization".
+
+#### Lifecycle Implementation
+
+1. The lifecycle will provide additional `-shared-cache-dir` and `-shared-cache-image` options where `-cache-dir` and `-cache-image` could be provided.
+2. Layers marked as `cache = true` and `cache-scope = "shared"` will be stored in either `-shared-cached-dir` or `-shared-cache-image`.
+3. If `-share-cached-dir` or `-share-cached-image` are not provide and a buildpack sets a layer to cache with scope `shared`, then it would be cached to the standard cache location (`-cache-dir`/`-cache-image`).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Requires that the buildpack author be continuous about providing this information for optimization purposes.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Generic Read/Write volume mounts (RFC PR [#85](https://github.com/buildpacks/rfcs/pull/85))
+    - The buildpack user must know what and where to mount things.
+    - Has safety concerns when writing to Linux mounts.
+    - Solution violates the spec dependent on use.
+    - Solution is specific to `pack`.
+    
+# Prior Art
+[prior-art]: #prior-art
+
+None
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+
+# Spec. Changes
+[spec-changes]: #spec-changes
+
+### Buildpacks Spec
+
+##### [Layer Content Metadata TOML](https://github.com/buildpacks/spec/blob/main/buildpack.md#layer-content-metadata-toml)
+```toml
+# whether this layer should be available to at launch (run-time)
+launch = false
+
+# whether this layer should be available to subsequent buildpacks
+# type: boolean
+build = false
+
+# whether this layer should be cached
+# boolean
+# type: boolean
+cache = false
+
+# type of cache (if `cache = true`)
+# type: string
+# values: app, shared
+# default: app
+cache-scope = "app"
+
+[metadata]
```

Comment:
I wonder if `metadata` gets problematic in this situation. The buildpack would need to be careful not to fight with itself about the contents of the metadata. 

E.G it would append new dependencies to the metadata if it adds deps to a shared layers (if it is recording that type of thing) instead of replacing the metadata wholesale with the list of deps relevant to a given build.